### ISSUE TITLE
Allow fzpz in fritzing-parts

### ISSUE THREAD

**Issue by KjellMorgenstern** [fritzing/fritzing-app#4018]
## Problem
Parts a currently structured in folders for fzp and layers.
This structure allows to reuse certain standardized footprints. However, custom parts that come with their own footprint and symbols are a very common use case. For these cases, the organization into folders and sharing of files causes complexity that is hard to manage and not necessary.

## Proposed Solution
Fritzing app should allow fzpz in fritzing-parts/fzpz in arbitrary subfolders

**Comment by failiz** [comment]
I agree with simplifying the organization requirements for parts that have all custom SVG files. However, I do not think we should store fzpz files in the repository as they are compressed and version control would be useless. Maybe we could create a folder for each file where the fpz and SVGs files are stored in plain text.

**Comment by vanepp** [comment]
I tend to agree with @failiz  here, I would prefer to keep the repository to uncompressed files. I think that this may have been what the contrib directories were intended for (or could be successfully repurposed for anyway.)  Here is the current repository layout, the contrib subtree is present but not apparently used

![1](https://user-images.githubusercontent.com/16514277/227600487-c93b36aa-abe2-482b-9749-536c0f2ec4ec.jpg)

![2](https://user-images.githubusercontent.com/16514277/227600592-b097180b-104c-44b9-8498-112edca744b7.jpg)

![3](https://user-images.githubusercontent.com/16514277/227600634-d1c11283-097c-4396-88c3-27265958be50.jpg)

a loader modification that would unpack the user's .fzpz file in to this tree may be the best answer here I think, as it leaves the repository uncompressed but perhaps does what Kjell is looking for? I'm not sure what implications parts in those directories have, I assume they are searched when looking for parts, but that may not have been implemented yet when development stopped.

**Comment by KjellMorgenstern** [comment]
The large number of files is a bit heavy on some file systems. 
It could be possible to use @failiz idea for the git repo, but still deploy Fritzing with zipped versions.
However, from a coding perspective, extracting parts into their own subdirectory (using the module id as folder name?) without any zipping at all, might be the easiest solution.

The current approach (core folder) has multiple downsides: 1. It is quite awkward and often difficult to keep files for a part in sync during editing. 2. Because parts share files, editing a part can always have side effects. 3. Unzipping a fzpz is often not enough to sort it into fritzing-parts. The file names might not be in the right pattern anymore.

Using the contrib folder for this might cause some trouble. Instead of fritzing-parts/fzpz, we could
use `fritzing-parts/core/**/{module_id}/{part_files}`
where part files would have the same filenames as in the fzpz (but ignore any folders used within the fzpz)

If there would be a usage of the contrib folder anytime in the future, the same pattern could apply (  `fritzing-parts/contrib/**/{module_id}/{part_files}` )

Fritzing would ignore the /**/ folder structure, that would be just for sorting during editing parts. Any user-facing categorization would use tags or other part properties.

**Comment by KjellMorgenstern** [comment]
Reviewing this again: fzpz might be a bit cleaner than spreading part files over core and svg directories, but I think this issue can also be solved with proper tools, like the checks in the CI pipeline.

 -> Low Priority or even Close this issue ?
